boring/src/pkey.rs

```diff
@@ -571,6 +589,18 @@ mod tests {
         assert!(pub_key.windows(10).any(|s| s == b"PUBLIC KEY"));
     }
 
+    #[test]
+    fn test_der_pkcs8() {
+        let key = include_bytes!("../test/key.der");
+        let key = PKey::private_key_from_der(key).unwrap();
+
+        let priv_key = key.private_key_to_der_pkcs8().unwrap();
+
+        // Check that this has the correct PKCS#8 version number and algorithm.
+        assert_eq!(hex::encode(&priv_key[4..=6]), "020100"); // Version 0
+        assert_eq!(hex::encode(&priv_key[9..=19]), "06092a864886f70d010101"); // Algorithm RSA/PKCS#1
```

Comment:
Derived from <https://asn1.io/PKI-inspector/> and its sample PKCS#8 private key.